sdk/storage/Azure.Storage.Blobs/src/BlobQuickQueryStream.cs

```diff
@@ -131,6 +131,14 @@ private async Task<int> ReadInternal(bool async, byte[] buffer, int offset, int
                     case Constants.QuickQuery.DataRecordName:
                         record.TryGetValue(Constants.QuickQuery.Data, out object byteObject);
                         byte[] bytes = (byte[])byteObject;
+
+                        if (_buffer != null)
+                        {
+                            ArrayPool<byte>.Shared.Return(_buffer);
+                        }
+
+                        _buffer = ArrayPool<byte>.Shared.Rent(bytes.Length);
```

Comment:
Consider doing `Math.Max(4MB, bytes.Length)` to rent a buffer that'll be big enough for most records

Reply:
Fixed in next iteration.